### PR TITLE
docs(testing): document automated testing systems

### PR DESCRIPTION
**Pull request by ace-n**
I felt like our automated testing docs were a bit light, so I 🦅 _took the liberty_ 🦅  of fleshing them out. 😄 

~**BLOCKED** on #347~

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+
+### Pipelines / Delivery System
+  
+The **delivery system** is covered by an automatic **End-to-End (E2E) Test**.
+
+This test uses Cloud Build to run the [setup/deployment script](/setup.sh), and checks that it completes without any errors.
+  
+These tests run against the `main` branch on a **nightly** basis.
+
+> **Note:** this is currently [in progress](https://github.com/GoogleCloudPlatform/emblem/issues/347)
```
Comment (grayside) [inline docs/testing.md:19]: If we're going to merge this in the docs, let's remove future things or hold this in waiting until it's true.

File: docs/testing.md [hunk 2]
```diff
@@ -1,3 +1,34 @@
+# Automated Tests
```
Comment (grayside) [inline docs/testing.md:1]: Thanks so much for adding this documentation!

**Review by grayside — DISMISSED** [review]
Thanks for getting this started! I've already got a testing.md file merged in docs, with a focus on contributor instructions on how to run tests.

I figure this content has several paths forward, a couple options I see:

* Rename to delivery.md, with the intent to expand into more deployment architecture content in follow-ups
* Rebase onto the merged testing.md, to explain more about automated testing systems after covering how to run tests

File: docs/testing.md (hunk 1 above)
Comment (ace-n) [reply docs/testing.md:19]: 👍 let's do the latter

**Comment by ace-n** [comment]
> Rebase onto the merged testing.md, to explain more about automated testing systems after covering how to run tests

This was my intent

File: docs/testing.md (hunk 1 above)
Comment (ace-n) [reply docs/testing.md]: I linked to the (optional) `installation-testing` directory instead.

I'll include a nightly build trigger (via Terraform) in there, via a separate PR.

**Review by grayside — APPROVED** [review]

**Comment by grayside** [comment]
LGTM. I will follow-up to rotate manual testing up and give the doc a top-level heading.

**Pull request merged**